community/io/src/main/java/org/neo4j/io/pagecache/monitoring/PageCacheMonitor.java

```diff
@@ -19,311 +19,13 @@
  */
 package org.neo4j.io.pagecache.monitoring;
 
-import java.io.File;
-import java.io.IOException;
-
-import org.neo4j.io.pagecache.PageSwapper;
-
 /**
- * A PageCacheMonitor receives a steady stream of events and data about what
- * the page cache is doing. Implementations of this interface should be as
- * efficient as possible, lest they severely slow down the page cache.
+ * The PageCacheMonitor exposes internal counters from the page cache.
+ * The data for these counters is sourced through the PageCacheTracer API.
  */
 public interface PageCacheMonitor
 {
     /**
-     * A FlushEvent implementation that does nothing.
-     */
-    public static final FlushEvent NULL_FLUSH_EVENT = new FlushEvent()
```

Comment:
Polymorphism is a property of the individual callsite. If you start two databases in the same JVM and configure them with different tracers, then you'll see it influence optimisation, but otherwise not.